service/user.js

```diff
@@ -0,0 +1,29 @@
+const User = require("./schemas/user");
+
+const createUser = ({ email, password }) => {
+  return User.create({ email, password });
+};
+
+const findUser = ({ email,token }) => {
+  return User.findOneAndUpdate({email, token});
+};
+
+const findUserAndUpdateToken = async (email, newToken) => {
+  return User.findOneAndUpdate(
+    { email },
+    { token: newToken },
+    { new: true } // Zwraca zaktualizowany dokument
+  );
+};
+
+
+const findUserByID = ({ id }) => {
```

Comment:
Podziel sobie te funkcje, czyli jak masz find to find, a jak robisz update, to zrób osobną funkcję.